### PR TITLE
Hw

### PR DESCRIPTION
**Pull request by RetKinf**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -58,6 +58,11 @@
             <artifactId>mapstruct</artifactId>
             <version>1.5.5.Final</version>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-security</artifactId>
+            <version>3.3.0</version>
```
Comment (Elena-Bruyako) [inline pom.xml:64]:
```suggestion
```

File: src/main/java/com/example/config/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+
+@Configuration
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/config/SecurityConfig.java:14]:
```suggestion
@Configuration
@RequiredArgsConstructor
@EnableMethodSecurity
```

File: src/main/java/com/example/config/SecurityConfig.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+public class SecurityConfig {
+    private final UserDetailsService userDetailsService;
+
+    public SecurityConfig(UserDetailsService userDetailsService) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/config/SecurityConfig.java:18]: remove

File: src/main/java/com/example/security/CustomUserDetailsService.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        return userRepository.findByEmail(email)
+                .orElseThrow(() -> new UsernameNotFoundException(
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/security/CustomUserDetailsService.java:18]:
```suggestion
                .orElseThrow(() -> new EntityNotFoundException(
```

File: src/main/java/com/example/service/UserServiceImpl.java [hunk 5]
```diff
@@ -24,6 +29,8 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto)
             );
         }
         User user = userMapper.toModel(requestDto);
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
+        user.getRoles().add(roleRepository.findByRole(RoleName.USER));
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/service/UserServiceImpl.java:33]:
```suggestion
        role = roleRepository.findByRole(RoleName.USER);
        user.setRole(Set.of(role));
```

File: src/main/resources/db/changelog/changes/04-create-users-roles-table.yaml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+                  name: user_id
+                  type: bigint
+                  constraints:
+                    nullable: false
+              - column:
+                  name: role_id
+                  type: bigint
+                  constraints:
+                    nullable: false
+        - addPrimaryKey:
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/04-create-users-roles-table.yaml:19]: remove PK please

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's fix some comments

File: src/main/resources/db/changelog/changes/05-insert-users.yaml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+      changes:
+        - insert:
+            tableName: users
+            columns:
+              - column:
+                  name: email
+                  value: "admin@example.com"
+              - column:
+                  name: password
+                  value: "$2a$10$r5Ih0Fdb/DATMcG41JKdiuBsrhslSH4qZ9wuDcj9qRpZwtUzDHLOa"
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/05-insert-users.yaml:14]: please add 
```suggestion
                  value: "$2a$10$r5Ih0Fdb/DATMcG41JKdiuBsrhslSH4qZ9wuDcj9qRpZwtUzDHLOa" # 123456
```
please add passwords in plaintext as comment

File: src/main/java/com/example/model/User.java [hunk 8]
```diff
@@ -5,19 +5,28 @@
...
 import org.hibernate.annotations.SQLDelete;
 import org.hibernate.annotations.SQLRestriction;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
 
 @Getter
 @Setter
 @Entity
 @SQLDelete(sql = "UPDATE users SET is_deleted = true WHERE id=?")
 @SQLRestriction("is_deleted=false")
```
Comment (okuzan) [inline src/main/java/com/example/model/User.java:27]:
```suggestion
@SQLDelete(sql = "UPDATE users SET is_deleted = TRUE WHERE id = ?")
@SQLRestriction("is_deleted = FALSE")
```

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/security/CustomUserDetailsService.java (hunk 4 above)
Comment (okuzan) [reply src/main/java/com/example/security/CustomUserDetailsService.java]: UsernameNotFound is a suitable choice here too, even more specific, which is good

**Pull request merged**